File: src/model/course-admin/CourseCardResponse.ts
export interface CourseCardResponse {
  courseId: string;
  title: string;
  slug: string;
  shortDescription: string;
  imageUrl?: string;
  isPublished: boolean;
  totalModules: number;
  publishedAt?: string;
  enrolledCount?: number;
  capacity?: number;
  instructorName?: string;
  showStudentCount?: boolean;
  showProgress?: boolean;
  progressPercentage?: number;
}

File: src/main/java/com/truongsonkmhd/unetistudy/dto/course_dto/CourseCardResponse.java
package com.truongsonkmhd.unetistudy.dto.course_dto;

import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CourseCardResponse {
    private UUID courseId;
    private String title;
    private String slug;
    private String shortDescription;
    private String imageUrl;

    private Boolean isPublished;
    private Integer totalModules;
    private LocalDateTime publishedAt;
    private Integer enrolledCount;
    private Integer capacity;
    private String instructorName;
}